Tests/UnitTests/PaymentsCoreTests/NetworkingTests/APIClient_Tests.swift

```diff
@@ -0,0 +1,189 @@
+            statusCode: 200,
+            responseString: mockSuccessResponse
+        )
+
+        URLProtocolMock.requestResponses.append(mockRequestResponse)
+
+        apiClient.fetch(endpoint: mockRequestResponse) { result, _ in
+            guard case .success(let response) = result else {
+                XCTFail("Expect success response")
+                return
+            }
+
+            XCTAssertEqual(response.accessToken, "TestToken")
+            expect.fulfill()
+        }
+        waitForExpectations(timeout: 1)
+    }
+
+    func testFetch_withAccessTokenFailureMockResponse_vendsUnknownError() {
+        let expect = expectation(description: "Get mock response for access token request")
+
+        let mockFailureResponse: String = """
+        {
+            "error": "unsupported_grant_type",
+            "error_description": "unsupported grant_type"
+        }
+        """
+
+        let mockRequestResponse = MockRequestResponse(
+            request: AccessTokenRequest(clientID: ""),
+            statusCode: 404,
+            responseString: mockFailureResponse
+        )
+
+        URLProtocolMock.requestResponses.append(mockRequestResponse)
+
+        apiClient.fetch(endpoint: mockRequestResponse) { result, _ in
+            guard case .failure(.unknown) = result else {
+                XCTFail("Expect NetworkingError.unknown for status code 404")
+                return
+            }
+
+            expect.fulfill()
+        }
+        waitForExpectations(timeout: 1)
+    }
+
+    func testFetch_withAccessTokenInvalidMockResponse_returnsDecodingError() {
+        let expect = expectation(description: "Get mock response for access token request")
+
+        let mockInvalidResponse: String = """
+        {
+            "test": "wrong response format"
+        }
+        """
+
+        let mockRequestResponse = MockRequestResponse(
+            request: AccessTokenRequest(clientID: ""),
+            statusCode: 200,
+            responseString: mockInvalidResponse
+        )
+
+        URLProtocolMock.requestResponses.append(mockRequestResponse)
+
+        apiClient.fetch(endpoint: mockRequestResponse) { result, _ in
+            guard case .failure(.decodingError) = result else {
+                XCTFail("Expect NetworkingError.decodingError for invalid response")
+                return
+            }
+
+            expect.fulfill()
+        }
+        waitForExpectations(timeout: 1)
+    }
+
+    func testFetch_withEmptyResponse_vendsSuccessfully() {
+        let expect = expectation(description: "Get empty response type for mock request")
+        let emptyRequest = MockEmptyRequestResponse()
+
+        URLProtocolMock.requestResponses.append(emptyRequest)
+
+        apiClient.fetch(endpoint: emptyRequest) { result, _ in
+            guard case .success = result else {
+                XCTFail("Expected successful empty response")
+                return
+            }
```

Comment:
This test should have a positive assertion on the result instead

Reply:
similar to below comment, since `.success` has an associated value we'd need to do `guard case` here.
We can do another positive assertion below this like `XCTAssertTrue(result is EmptyResponse)` but that'd always be true due to `associatedtype` of `EmptyRequestResponseMock` is already `EmptyResponse`